### PR TITLE
Add a WYSIWYG-enabled body field to products

### PR DESCRIPTION
**Pull request by mglaman**
Adds the WYSIWYG body field. However, I wasn't sure if the description field was to stay on the Commerce Product Type.

### CODE REVIEW

File: modules/product/src/CommerceProductTypeInterface.php [hunk 1]
```diff
@@ -14,22 +14,4 @@
...
-   * Sets the description of the product type.
-   *
-   * @param string $description
-   *   The new description.
-   *
-   * @return $this
-   */
-  public function setDescription($description);
-
 }
```
Comment (mglaman) [inline modules/product/src/CommerceProductTypeInterface.php:35]: Should this have stayed, @bojanz, and only CommerceProduct manipulated? Still getting a grip on D8 entities.

**Comment by bojanz** [comment]
The product type shouldn't be touched, we want them to have a description (it's shown below the product type label on the product/add page).

**Comment by mglaman** [comment]
@bojanz this has been fixed to not fiddle with product type descriptions.

File: modules/product/commerce_product.module [hunk 2]
```diff
@@ -100,3 +102,54 @@ function commerce_product_theme() {
     ),
   );
 }
+
+/**
+ * Adds the default body field to a custom block type.
```
Comment (bojanz) [inline modules/product/commerce_product.module]: Still mentions blocks.

File: modules/product/commerce_product.module [hunk 3]
```diff
@@ -100,3 +102,54 @@ function commerce_product_theme() {
...
+ *
+ * @param string $commerce_product_type_id
+ *   CommerceProductInterface instance.
```
Comment (bojanz) [inline modules/product/commerce_product.module]: Nope, this is not a CommerceProductInterface instance, this is the id of the product type.

File: modules/product/commerce_product.module [hunk 4]
```diff
@@ -100,3 +102,54 @@ function commerce_product_theme() {
...
+ * @param string $label
+ *   (optional) The label for the body instance. Defaults to 'Body'
+ *
+ * @return array()
```
Comment (bojanz) [inline modules/product/commerce_product.module]: I think array() should be just array

File: modules/product/src/Entity/CommerceProductType.php [hunk 5]
```diff
@@ -83,4 +84,15 @@ public function setDescription($description) {
...
 
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(EntityStorageInterface $storage, $update = TRUE) {
+    parent::postSave($storage, $update);
+
+    $label = $this->get('create_body_label');
+    commerce_product_add_body_field($this->id, $label);
+
```
Comment (bojanz) [inline modules/product/src/Entity/CommerceProductType.php]: Extra newline.

**Comment by mglaman** [comment]
Should be updated per your initial code review @bojanz

File: modules/product/src/Entity/CommerceProductType.php [hunk 6]
```diff
@@ -83,4 +84,14 @@ public function setDescription($description) {
     return $this;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(EntityStorageInterface $storage, $update = TRUE) {
+    parent::postSave($storage, $update);
+
+    $label = $this->get('create_body_label');
```
Comment (bojanz) [inline modules/product/src/Entity/CommerceProductType.php]: You're using the create_body_label setting, but it doesn't exist.
Let's add it, and let's add "create_body" too while we're at it.

File: modules/product/commerce_product.module [hunk 7]
```diff
@@ -100,3 +102,54 @@ function commerce_product_theme() {
     ),
   );
 }
+
+/**
+ * Adds the default body field to a Commerce Product type.
```
Comment (bojanz) [inline modules/product/commerce_product.module]: Let's change this to "Adds the default body field to a product type."

**Comment by mglaman** [comment]
Implemented create_body and create_body_label, updated commerce_product_add_body_field description.

File: modules/product/src/Entity/CommerceProductType.php [hunk 8]
```diff
@@ -83,4 +102,19 @@ public function setDescription($description) {
...
+  public function postSave(EntityStorageInterface $storage, $update = TRUE) {
+    parent::postSave($storage, $update);
+
+    // Create a body if the create_body property is true and we're not in
+    // the syncing process.
+    if ($this->get('create_body') && !$this->isSyncing()) {
+      $label = $this->get('create_body_label');
+      commerce_product_add_body_field($this->id, $label);
+    }
+
```
Comment (bojanz) [inline modules/product/src/Entity/CommerceProductType.php]: Again that extra newline :)

File: modules/product/src/Entity/CommerceProductType.php [hunk 9]
```diff
@@ -69,6 +70,24 @@ class CommerceProductType extends ConfigEntityBundleBase implements CommerceProd
   protected $description;
 
   /**
+   * Indicates whether a Body field should be created for this product type.
```
Comment (bojanz) [inline modules/product/src/Entity/CommerceProductType.php]: Why is Body capitalized here?

File: modules/product/commerce_product.module [hunk 10]
```diff
@@ -100,3 +102,54 @@ function commerce_product_theme() {
...
+ *
+ * @param string $commerce_product_type_id
+ *   Id of the Commerce Product type.
+ * @param string $label
+ *   (optional) The label for the body instance. Defaults to 'Body'
+ *
+ * @return array
+ *   Body field instance.
+ */
+function commerce_product_add_body_field($commerce_product_type_id, $label = 'Body') {
```
Comment (bojanz) [inline modules/product/commerce_product.module]: Let's go with $product_type_id for brevity.

**Comment by mglaman** [comment]
@bojanz I know you mentioned in IRC needing to update the schema, however not seeing what would have to be updated/remove/added.

**Comment by bojanz** [comment]
Nevermind, I was mistaken.

**Pull request merged**